operations/agent_monitor.go

```diff
@@ -402,6 +409,12 @@ func (m *monitor) runAgent(ctx context.Context, retry util.RetryArgs) error {
 func (m *monitor) run(ctx context.Context) {
 	for {
 		if err := util.RetryWithArgs(ctx, func() (bool, error) {
+			if util.StringSliceContains(evergreen.ProviderEc2Type, m.provider) {
+				if util.SpotHostWillTerminateSoon() {
+					return true, errors.New("spot host terminating soon, not starting a new agent")
```

Comment:
This should return false.

Reply:
I returned true deliberately. Sam was concerned that this kind of logic in the monitor is risky, since exiting the monitor could strand hosts. Instead my thought was to keep retrying, but to noop the retry.

Reply:
I don't know what you mean by strand hosts in this case. I don't think this is wrong, but it seems like retrying is unnecessary since the host is imminently going to be terminated.

Reply:
I was being extra careful. If for some reason `util.SpotHostWillTerminateSoon` returns `true`, but the host isn't actually going to terminate, the monitor won't actually exit. Instead it will retry. I don't think there's risk in it retrying, since the action is entirely local.